src/allmydata/test/test_storage.py

```diff
@@ -3309,21 +3318,21 @@ def _then(results):
         d.addCallback(lambda ignored:
             sdmfw.get_checkstring())
         d.addCallback(lambda checkstring:
-            self.failUnlessEqual(checkstring, self._expected_checkstring))
+            self.assertThat(checkstring, Equals(self._expected_checkstring)))
         d.addCallback(lambda ignored:
             sdmfw.finish_publishing())
         def _then_again(results):
-            self.failUnless(results[0])
+            self.assertTrue(results[0])
             read = self.ss.slot_readv
-            self.failUnlessEqual(read(b"si1", [0], [(1, 8)]),
+            self.assertThat(read(b"si1", [0], [(1, 8)]),
                                  {0: [struct.pack(">Q", 1)]})
-            self.failUnlessEqual(read(b"si1", [0], [(9, len(data) - 9)]),
-                                 {0: [data[9:]]})
+            self.assertThat(read(b"si1", [0], [(9, len(data) - 9)]),
+                                 Equals({0: [data[9:]]}))
         d.addCallback(_then_again)
         return d
 
 
-class Stats(unittest.TestCase):
+class Stats(SyncTestCase):
```

Comment:
setUp and tearDown need to call the super implementations

Reply:
Resolved